### PR TITLE
Fix #2296:Add timezone while displaying time in challenge/phase dates

### LINKED ISSUE #2296: Add Timezone while displaying time in challenges
**Current Behaviour**

The challenge/challenge phase start and end date are shown without the timezone which sometimes confuses the user.

**Expected Behaviour**
The timezone should be visible on the UI.

For reference: #1766

### PR DESCRIPTION
**Pull request by sarthak212**
Fix #2296 
@RishabhJain2018 please review.

### CODE REVIEW

File: frontend/src/js/controllers/challengeCtrl.js [hunk 1]
```diff
@@ -423,6 +423,13 @@
                         vm.phases.results[i].showPrivate = true;
                     }
                 }
+                for(var i=0;i<vm.phases.results.length;i++){
+                    var qwerty=new Date(vm.phases.results[i].start_date).toString();
+                    vm.phases.results[i].start_zone=qwerty.split('(')[1].split(')')[0];
```
Comment (galipremsagar) [inline frontend/src/js/controllers/challengeCtrl.js]: Please refrain from doing such extraction..Please make use of moment js which we are already using in our project.
FYI: Extracting timezone is pretty straight forward with moment: https://momentjs.com/timezone/

File: frontend/src/js/controllers/challengeListCtrl.js [hunk 2]
```diff
@@ -83,7 +87,13 @@
                             } else {
                                 vm.upcomingList[i].isLarge = "";
                             }
+                            
+                            var qwerty=new Date(vm.upcomingList[i].start_date).toString();
+                            vm.upcomingList[i].start_zone=qwerty.split('(')[1].split(')')[0];
```
Comment (galipremsagar) [inline frontend/src/js/controllers/challengeListCtrl.js]: same as above.

File: frontend/src/js/controllers/challengeListCtrl.js [hunk 3]
```diff
@@ -115,6 +125,12 @@
                                     } else {
                                         vm.pastList[i].isLarge = "";
                                     }
+
+                                    var qwerty=new Date(vm.pastList[i].start_date).toString();
+                                    vm.pastList[i].start_zone=qwerty.split('(')[1].split(')')[0];
```
Comment (galipremsagar) [inline frontend/src/js/controllers/challengeListCtrl.js]: same as above.

**Comment by sarthak212** [comment]
@galipremsagar @RishabhJain2018 U can see the zones in screenshot .
zone1
zone2

**Comment by sarthak212** [comment]
@galipremsagar applied the changes please review.

File: frontend/src/js/controllers/challengeListCtrl.js [hunk 4]
```diff
@@ -52,6 +52,11 @@
                     } else {
                         vm.currentList[i].isLarge = "";
                     }
+                    var timezone=moment.tz.guess();
```
Comment (galipremsagar) [inline frontend/src/js/controllers/challengeListCtrl.js]:
```suggestion

```
the timezone of the client can be calculated only once, not needed to be done repeatedly. Moving out of loop.

File: frontend/src/js/controllers/challengeListCtrl.js [hunk 5]
```diff
@@ -52,6 +52,11 @@
...
+                    var offset=new Date(vm.currentList[i].start_date).getTimezoneOffset();
```
Comment (galipremsagar) [inline frontend/src/js/controllers/challengeListCtrl.js]:
```suggestion
                    var offset = new Date(vm.currentList[i].start_date).getTimezoneOffset();
```
Indent :)

File: frontend/src/js/controllers/challengeListCtrl.js [hunk 6]
```diff
@@ -52,6 +52,11 @@
...
+                    vm.currentList[i].start_zone= moment.tz.zone(timezone).abbr(offset);
```
Comment (galipremsagar) [inline frontend/src/js/controllers/challengeListCtrl.js]:
```suggestion
                    vm.currentList[i].start_zone = moment.tz.zone(timezone).abbr(offset);
```

File: frontend/src/js/controllers/challengeListCtrl.js [hunk 7]
```diff
@@ -52,6 +52,11 @@
...
+                    offset=new Date(vm.currentList[i].end_date).getTimezoneOffset();
```
Comment (galipremsagar) [inline frontend/src/js/controllers/challengeListCtrl.js]:
```suggestion
                    offset = new Date(vm.currentList[i].end_date).getTimezoneOffset();
```

File: frontend/src/js/controllers/challengeListCtrl.js [hunk 8]
```diff
@@ -52,6 +52,11 @@
...
+                    vm.currentList[i].end_zone=moment.tz.zone(timezone).abbr(offset);
```
Comment (galipremsagar) [inline frontend/src/js/controllers/challengeListCtrl.js]:
```suggestion
                    vm.currentList[i].end_zone = moment.tz.zone(timezone).abbr(offset);
```

File: frontend/src/js/controllers/challengeListCtrl.js [hunk 9]
```diff
@@ -83,7 +88,14 @@
                             } else {
                                 vm.upcomingList[i].isLarge = "";
                             }
-
+                            
+                            var timezone=moment.tz.guess();
```
Comment (galipremsagar) [inline frontend/src/js/controllers/challengeListCtrl.js]:
```suggestion
```
Please move **var timezone = moment.tz.guess();** to before start of forloop.

File: frontend/src/js/controllers/challengeListCtrl.js [hunk 10]
```diff
@@ -83,7 +88,14 @@
...
+                            var offset=new Date(vm.upcomingList[i].start_date).getTimezoneOffset();
```
Comment (galipremsagar) [inline frontend/src/js/controllers/challengeListCtrl.js]:
```suggestion
                            var offset = new Date(vm.upcomingList[i].start_date).getTimezoneOffset();
```

File: frontend/src/js/controllers/challengeListCtrl.js [hunk 11]
```diff
@@ -83,7 +88,14 @@
...
+                            vm.upcomingList[i].start_zone= moment.tz.zone(timezone).abbr(offset);
```
Comment (galipremsagar) [inline frontend/src/js/controllers/challengeListCtrl.js]:
```suggestion
                            vm.upcomingList[i].start_zone = moment.tz.zone(timezone).abbr(offset);
```

File: frontend/src/js/controllers/challengeListCtrl.js [hunk 12]
```diff
@@ -83,7 +88,14 @@
...
+                            offset=new Date(vm.upcomingList[i].end_date).getTimezoneOffset();
```
Comment (galipremsagar) [inline frontend/src/js/controllers/challengeListCtrl.js]:
```suggestion
                            offset = new Date(vm.upcomingList[i].end_date).getTimezoneOffset();
```

File: frontend/src/js/controllers/challengeListCtrl.js [hunk 13]
```diff
@@ -83,7 +88,14 @@
...
+                            vm.upcomingList[i].end_zone=moment.tz.zone(timezone).abbr(offset);
```
Comment (galipremsagar) [inline frontend/src/js/controllers/challengeListCtrl.js]:
```suggestion
                            vm.upcomingList[i].end_zone = moment.tz.zone(timezone).abbr(offset);
```

File: frontend/src/js/controllers/challengeListCtrl.js [hunk 14]
```diff
@@ -115,6 +127,13 @@
                                     } else {
                                         vm.pastList[i].isLarge = "";
                                     }
+
```
Comment (galipremsagar) [inline frontend/src/js/controllers/challengeListCtrl.js]:
```suggestion
```

File: frontend/src/js/controllers/challengeListCtrl.js [hunk 15]
```diff
@@ -115,6 +127,13 @@
...
+                                    var timezone=moment.tz.guess();
```
Comment (galipremsagar) [inline frontend/src/js/controllers/challengeListCtrl.js]:
```suggestion
```
Please move **var timezone = moment.tz.guess();** to before forloop

File: frontend/src/js/controllers/challengeListCtrl.js [hunk 16]
```diff
@@ -115,6 +127,13 @@
...
+                                    var offset=new Date(vm.pastList[i].start_date).getTimezoneOffset();
```
Comment (galipremsagar) [inline frontend/src/js/controllers/challengeListCtrl.js]:
```suggestion
                                    var offset = new Date(vm.pastList[i].start_date).getTimezoneOffset();
```

File: frontend/src/js/controllers/challengeListCtrl.js [hunk 17]
```diff
@@ -115,6 +127,13 @@
...
+                                    vm.pastList[i].start_zone= moment.tz.zone(timezone).abbr(offset);
```
Comment (galipremsagar) [inline frontend/src/js/controllers/challengeListCtrl.js]:
```suggestion
                                    vm.pastList[i].start_zone = moment.tz.zone(timezone).abbr(offset);
```

File: frontend/src/js/controllers/challengeListCtrl.js [hunk 18]
```diff
@@ -115,6 +127,13 @@
...
+                                    offset=new Date(vm.pastList[i].end_date).getTimezoneOffset();
```
Comment (galipremsagar) [inline frontend/src/js/controllers/challengeListCtrl.js]:
```suggestion
                                    offset = new Date(vm.pastList[i].end_date).getTimezoneOffset();
```

File: frontend/src/js/controllers/challengeListCtrl.js [hunk 19]
```diff
@@ -115,6 +127,13 @@
...
+                                    vm.pastList[i].end_zone=moment.tz.zone(timezone).abbr(offset);
```
Comment (galipremsagar) [inline frontend/src/js/controllers/challengeListCtrl.js]:
```suggestion
                                    vm.pastList[i].end_zone = moment.tz.zone(timezone).abbr(offset);
```

File: frontend/src/js/controllers/challengeCtrl.js [hunk 20]
```diff
@@ -423,6 +423,14 @@
...
                     }
                 }
+                for(var i=0;i<vm.phases.results.length;i++){
+                    var timezone=moment.tz.guess();
+                    var offset=new Date(vm.phases.results[i].start_date).getTimezoneOffset();
+                    vm.phases.results[i].start_zone= moment.tz.zone(timezone).abbr(offset);
+                    offset=new Date(vm.phases.results[i].end_date).getTimezoneOffset();
+                    vm.phases.results[i].end_zone=moment.tz.zone(timezone).abbr(offset);
+                }
+
```
Comment (galipremsagar) [inline frontend/src/js/controllers/challengeCtrl.js]: Why have another for loop? You could do the same inside above for-loop. Extract the zip and Please apply my path to this ChallenegeCtrl.js file. 
[merge_into_single_for_loop_.patch.zip](https://github.com/Cloud-CV/EvalAI/files/3161441/merge_into_single_for_loop_.patch.zip)

**Comment by sarthak212** [comment]
@galipremsagar Resolve conflict and add changes as you mentioned. Please review

**Comment by sarthak212** [comment]
@galipremsagar @RishabhJain2018 Please review.

**Comment by galipremsagar** [comment]
Looks good to me. Perfect, Thanks for the changes. @RishabhJain2018 this can be merged.

**Comment by galipremsagar** [comment]
@sarthak212 please update your branch for the merging to happen

**Comment by RishabhJain2018** [comment]
LGTM :) Thanks @sarthak212 for adding this 🎉

**Comment by RishabhJain2018** [comment]
Also, @sarthak212 can you please add the timezone here as well https://github.com/Cloud-CV/EvalAI/blob/master/frontend/src/views/web/challenge/submission.html#L146

**Comment by sarthak212** [comment]
@RishabhJain2018 @galipremsagar make changes as you suggested and update the branch from latest commit.

**Pull request merged**